lib/dataset-tools/augmentation/replace_parameters.js

```diff
@@ -330,7 +330,10 @@ module.exports = class ParameterReplacer {
     _transformValue(sentenceValue, programValue, arg) {
         if (this._requotable) {
             if (arg && arg.name === 'id' && arg.type.isEntity && arg.type.type.endsWith(':Hotel')) {
-                sentenceValue = sentenceValue.slice(0, sentenceValue.indexOf(',')).trim();
+                if (sentenceValue.startsWith('the '))
+                    sentenceValue = sentenceValue.slice('the '.length).trim();
+                if (sentenceValue.indexOf(',') !== -1)
+                    sentenceValue = sentenceValue.slice(0, sentenceValue.indexOf(',')).trim();
```

Comment:
This does not seem correct in general. Also, `,` is not a good word so it will never show up.

Reply:
American English GoodWord accepts commas as well so that we can have locations  like "washington , dc" in params.
Some values from TripAdvisor contained repetitive/ excessive fields like " city, city, country" or "area, city, town, country". This is to clean those up.

Reply:
It's not a good heuristic to do for everything then. Put it in make-string-datasets